pkg/flow/flowtest/commands.go

```diff
@@ -0,0 +1,136 @@
+package flowtest
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/grafana/agent/component"
+	"github.com/grafana/agent/pkg/flow"
+	"github.com/grafana/dskit/backoff"
+	"rsc.io/script"
+)
+
+func createCommands(f *flow.Flow, opts *options) map[string]script.Cmd {
+	cmds := script.DefaultCmds()
+	registerControllerCommands(f, cmds)
+
+	for name, cmd := range opts.cmds {
+		cmds[name] = cmd
+	}
+
+	return cmds
+}
+
+func registerControllerCommands(f *flow.Flow, cmds map[string]script.Cmd) {
+	cmds["controller.load"] = script.Command(
```

Comment:
How would testing of two agents work? for example a proxy scenario or a cluster of agents?